tests/drivers/uart/uart_basic_api/src/test_uart_config.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * @addtogroup t_uart_basic
+ * @{
+ * @defgroup t_uart_config test_uart_config
+ * @brief TestPurpose: verify UART configure API settings
+ * @details
+ * - Test Steps
+ *   - Configure: test_uart_configure( )
+ *   - Configure Get: test_uart_config_get( )
+ * - Expected Results
+ *   -# When test UART CONFIG Configure, the value of configurations actually
+ *      set will be equal to the original configuration values (from device
+ *      tree or run-time configuration to modify those loaded initially from
+ *      device tree)
+ *   -# When test UART CONFIG Configure Get, the app will get/retrieve the
+ *      value of configurations stored at location and to be passed to UART
+ *      CONFIG Configure
+ * @}
+ */
+
+#include "test_uart.h"
+
+/* test UART configure (set configuration) */
+static int test_configure(void)
+{
+	const struct uart_config uart_cfg = {
+		.baudrate = 115200,
+		.parity = UART_CFG_PARITY_NONE,
+		.stop_bits = UART_CFG_STOP_BITS_1,
+		.data_bits = UART_CFG_DATA_BITS_8,
+		.flow_ctrl = UART_CFG_FLOW_CTRL_NONE
+	};
+	struct device *uart_dev = device_get_binding(UART_DEVICE_NAME);
+
+	if (!uart_dev) {
+		TC_PRINT("Cannot get UART device\n");
+		return TC_FAIL;
+	}
+
+	/* Verify configure() - set device configuration using data in cfg */
+	int ret = uart_configure(uart_dev, &uart_cfg);
+
+	/* 0 if successful, - error code otherwise */
+	if (ret == 0) {
+		return TC_PASS;
+	} else {
+		return TC_FAIL;
+	}
```

Comment:
Nit: could do `return (ret == 0) ? TC_PASS : TC_FAIL;`

Reply:
I updated this. Much cleaner, thanks!